test/stateful_dataloader/test_state_dict.py

```diff
@@ -1511,25 +1512,26 @@ def load_state_dict(self, state_dict):
 class CountIterCallsIter(torch.utils.data.IterableDataset):
     def __init__(self, length):
         self.length = length
-        self.iter_calls = 0
+        self.local_iter_calls = 0
+        self.prev_state_iter_calls = 0
         self.items = []
 
     def __iter__(self):
         self.items = list(range(self.length))
-        self.iter_calls += 1
+        self.local_iter_calls += 1
         return self
 
     def __next__(self):
         if len(self.items) > 0:
-            self.items.popleft()
+            return self.items.pop(0)
         else:
             raise StopIteration
 
     def state_dict(self):
-        return {"iter_calls": self.iter_calls, "items": deepcopy(self.items)}
+        return {"iter_calls": self.local_iter_calls + self.prev_state_iter_calls, "items": deepcopy(self.items)}
 
     def load_state_dict(self, state_dict):
-        self.iter_calls = state_dict["iter_calls"]
+        self.prev_state_iter_calls = state_dict["iter_calls"]
         self.items = state_dict["items"]
```

Comment:
Let's revert this, except delete the `self.iter_calls` in load_state_dict and just use it to track calls in the test